pkg/scheduler/framework/plugins/dynamicresources/structured/namedresources/namedresourcesmodel_test.go

```diff
@@ -0,0 +1,294 @@
+		Instances: []InstanceAllocation{{
+			Instance: &resourceapi.NamedResourcesInstance{
+				Name: instance1,
+			},
+		}},
+	}
+
+	instance2 := "instance-2"
+	twoInstances := Model{
+		Instances: []InstanceAllocation{
+			{
+				Instance: &resourceapi.NamedResourcesInstance{
+					Name: instance1,
+				},
+			},
+			{
+				Instance: &resourceapi.NamedResourcesInstance{
+					Name: instance2,
+				},
+			},
+		},
+	}
+
+	testcases := map[string]struct {
+		model    Model
+		filter   *resourceapi.NamedResourcesFilter
+		requests []*resourceapi.NamedResourcesRequest
+
+		expectCreateErr   string
+		expectAllocation  []string
+		expectAllocateErr string
+	}{
+		"empty": {},
+
+		"broken-filter": {
+			filter: filterBrokenType,
+
+			expectCreateErr: "compile class filter CEL expression: must evaluate to bool",
+		},
+
+		"broken-request": {
+			requests: []*resourceapi.NamedResourcesRequest{requestBrokenType},
+
+			expectCreateErr: "compile request CEL expression: must evaluate to bool",
+		},
+
+		"no-resources": {
+			filter:   filterAny,
+			requests: []*resourceapi.NamedResourcesRequest{requestAny},
+
+			expectAllocateErr: "insufficient resources",
+		},
+
+		"okay": {
+			model:    oneInstance,
+			filter:   filterAny,
+			requests: []*resourceapi.NamedResourcesRequest{requestAny},
+
+			expectAllocation: []string{instance1},
+		},
+
+		"filter-mismatch": {
+			model:    oneInstance,
+			filter:   filterNone,
+			requests: []*resourceapi.NamedResourcesRequest{requestAny},
+
+			expectAllocateErr: "insufficient resources",
+		},
+
+		"request-mismatch": {
+			model:    oneInstance,
+			filter:   filterAny,
+			requests: []*resourceapi.NamedResourcesRequest{requestNone},
+
+			expectAllocateErr: "insufficient resources",
+		},
+
+		"many": {
+			model:    twoInstances,
+			filter:   filterAny,
+			requests: []*resourceapi.NamedResourcesRequest{requestAny, requestAny},
+
+			expectAllocation: []string{instance1, instance2},
+		},
+
+		"too-many": {
+			model:    oneInstance,
+			filter:   filterAny,
+			requests: []*resourceapi.NamedResourcesRequest{requestAny, requestAny},
+
+			expectAllocateErr: "insufficient resources",
+		},
+
+		"filter-evaluation-error": {
+			model:    oneInstance,
+			filter:   filterBrokenEvaluation,
+			requests: []*resourceapi.NamedResourcesRequest{requestAny},
+
+			expectAllocateErr: "evaluate filter CEL expression: no such key: no-such-attribute",
+		},
+
+		"request-evaluation-error": {
+			model:    oneInstance,
+			filter:   filterAny,
+			requests: []*resourceapi.NamedResourcesRequest{requestBrokenEvaluation},
+
+			expectAllocateErr: "evaluate request CEL expression: no such key: no-such-attribute",
+		},
+	}
+
+	for name, tc := range testcases {
+		t.Run(name, func(t *testing.T) {
+			tCtx := ktesting.Init(t)
+
+			controller, allocateErr := NewClaimController(tc.filter, tc.requests)
+			if allocateErr != nil {
+				if tc.expectCreateErr == "" {
+					tCtx.Fatalf("unexpected create error: %v", allocateErr)
```

Comment:
`require.Error` would not have handled the "no error expected" case, so some additional if checks would have been needed and therefore I am writing out the checks manually here.

I tend to prefer `require + assert` whenever they are usable because that reduces the amount of test code and (IMHO) is more readable.